source/extensions/filters/listener/http_inspector/http_protocol_header.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace HttpInspector {
+
+/**
+ * Http2 magic string.
+ */
+constexpr char HTTP2_CONNECTION_PREFACE[] = "\x50\x52\x49\x20\x2a\x20\x48\x54\x54\x50\x2f\x32\x2e"
+                                            "\x30\x0d\x0a\x0d\x0a\x53\x4d\x0d\x0a\x0d\x0a";
+constexpr uint32_t HTTP2_CONNECTION_PREFACE_LEN = 24;
+
+const std::vector<absl::string_view> HTTP_METHODS{
```

Comment:
put in a ConstSingleton to avoid static initialization fiasco, use absl::flat_hash_set

Reply:
It seems to be duplicate with the `Http::Headers`. I will move these value into the inspector as a static const data member. WDYT?

Reply:
It seems to be duplicate with `Http::Headers`. I moved them into inspector. Do you think if we still need this file?

Reply:
sounds reasonable.